### PR TITLE
feat: auto dep resolution (prototype)

### PR DESCRIPTION
**Pull request by deryrahman**
NB: this prototype is on top of binary deprecation PR

### CODE REVIEW

File: plugin/plugin_service.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	"context"
+
+	"github.com/goto/optimus/internal/models"
+)
+
+type (
+	ResourceURN string
+	Name        string
+	TaskConfig  map[string]string
+	Assets      map[string]string
```
Comment (sbchaos) [inline plugin/plugin_service.go:13]: We can skip defining separate models.

File: sdk/plugin/plugin.go [hunk 2]
```diff
@@ -49,6 +69,9 @@ type Info struct {
 	PluginType  Type  `yaml:",omitempty"`
 	PluginMods  []Mod `yaml:",omitempty"`
 
+	AssetParser            AssetParser
```
Comment (sbchaos) [inline sdk/plugin/plugin.go:72]: Where will this be defined, in the yaml for the plugin or at the Optimus side

File: core/job/service/job_service.go [hunk 3]
```diff
@@ -76,6 +78,16 @@ type PluginService interface {
 	GenerateDependencies(ctx context.Context, taskName job.TaskName, svcAcc, query string, destinationURN job.ResourceURN) ([]job.ResourceURN, error)
 }
 
+type (
+	GenerateUpstreamFunc     func(assets p.Assets) []p.ResourceURN
+	ConstructDestinationFunc func(config p.TaskConfig) p.ResourceURN
```
Comment (sbchaos) [inline core/job/service/job_service.go:83]: Can we return the actual values instead of func ?

File: sdk/plugin/plugin.go (hunk 2 above)
Comment (deryrahman) [reply sdk/plugin/plugin.go:72]: this will be defined in yaml for the plugin, I forgot to write the yaml tagstruct there

**Comment by deryrahman** [comment]
Close this PR, the implementation can be seen on this PR #157

**Pull request closed without merging**